### PR TITLE
Molality conversion and removing undefined enum

### LINKED ISSUE #103: Add Molality Units
#### Description:

These are useful for ground investigations (specifically the contaminants) - e.g. ,mol/kg. Not to be confused with Molarity which is mol/m<sup>3</sup> (per unit volume).

This will be merged in with UnitsNet on this PR:
https://github.com/angularsen/UnitsNet/pull/1341

### PR DESCRIPTION
**Pull request by Martian42**
### NOTE: Depends on 

 ### Issues addressed by this PR

 Closes #103 

 

 ### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EkD9lu1u_fdJqv5CkRwt_-kBFLE3Rc4rHBai6uZ2TDLiAA?e=jYnmAR
 ### Changelog

 ### Additional comments

Undefined are removed in latest UnitsNet version.

### CODE REVIEW

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
@Martian42 removing `undefined` is out of scope to the original issue and requires a broader discussion if UnitsNet have changed their support of `undefined` options. Can you add them back in for now and let this PR focus solely on the issue in #103

File: Units_oM/Units_oM.csproj [hunk 1]
```diff
@@ -19,5 +19,9 @@
     <Exec Command="xcopy &quot;$(TargetDir)$(TargetFileName)&quot;  &quot;$(ProgramData)\BHoM\Assemblies&quot; /Y" />
   </Target>
 
+  <ItemGroup>
```
[the remark this answers is not in the document]
Comment (Martian42) [reply Units_oM/Units_oM.csproj]: Thanks Fraser, UnitsNet removed from oM in the latest push.

**Review by FraserGreenroyd — APPROVED** [review]
Code reviewed and looks good, great work @Martian42 😄

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

File: Units_Engine/Convert/Molality/Molality.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            return double.NaN;
+        }
+
+        /***************************************************/
+
+        [Description("Convert SI units (molePerKilogram) into another duration unit.")]
+        [Input("molePerKilograms", "The number of molePerKilograms to convert.")]
+        [Input("unit", "The unit to convert to. This can be a string, or you can use the BHoM Enum MolalityUnit.")]
+        [Output("molality", "The equivalent quantity defined in the specified unit.")]
+        public static double ToMolality(this double molality, object unit)
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Molality/Molality.cs:71]:
```suggestion
        public static double ToMolality(this double molePerKilograms, object unit)
```

File: Units_Engine/Convert/Molality/Molality.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        {
+            if (Double.IsNaN(molality) || Double.IsInfinity(molality))
+            {
+                Compute.RecordError("Quantity is not a real number.");
+                return double.NaN;
+            }
+
+            UN.QuantityValue qv = molality;
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Molality/Molality.cs:79]:
```suggestion
            if (Double.IsNaN(molePerKilograms) || Double.IsInfinity(molePerKilograms))
            {
                Compute.RecordError("Quantity is not a valid number.");
                return double.NaN;
            }

            UN.QuantityValue qv = molePerKilograms;
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Based on compliance checks

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer
@BHoMBot check null-handling
@BHoMBot check serialisation

**Review by FraserGreenroyd — APPROVED** [review]
All looks good now the compliance is passing 😄

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**